### PR TITLE
feat(issue platform): Create grouptype table

### LINKED ISSUE #43626: Create grouptype dataclass
~~Create a migration for a new table to store the [group type data](https://github.com/getsentry/sentry/blob/master/src/sentry/types/issues.py) currently using enums.~~

~~This migration should also backfill the existing group types.~~

~~See https://github.com/getsentry/sentry/issues/43484 for more context and suggested columns.~~

Create a grouptype dataclass in place of the existing enums / dicts to store new issue type data.

### PR DESCRIPTION
**Pull request by ceorourke**
Create a new table `sentry_grouptype` to store new issue types.

Fixes #43626

### CODE REVIEW

File: src/sentry/models/grouptype.py [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+
+
+@region_silo_only_model
+class GroupType(Model):
+    __include_in_export__ = False
+
+    slug = models.CharField(max_length=254, unique=True)
+    description = models.CharField(max_length=254, unique=True)
+    category = BoundedPositiveIntegerField(
+        choices=[(category.value, category.name.lower()) for category in GroupCategory],
```
Comment (ceorourke) [inline src/sentry/models/grouptype.py:14]: Is this safe to do? I saw a similar usage in the `GroupTombstone` table's `level` column.

Comment (markstory) [reply src/sentry/models/grouptype.py:14]: This defines column metadata that django can use to do validation. It looks like your goals include dynamic category values that aren't defined in application code, which this wouldn't accommodate.

File: src/sentry/models/grouptype.py [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+        default=GroupCategory.ERROR.value,
+    )
+    ignore_limit = models.IntegerField(default=3, null=False)
+
+    class Meta:
+        app_label = "sentry"
+        db_table = "sentry_grouptype"
+        unique_together = (("slug", "description"),)
```
Comment (markstory) [inline src/sentry/models/grouptype.py:22]: Do you want to allow duplicate slugs if the descriptions are different or should slugs always be unique?

**Comment by markstory** [comment]
How/when does data get deleted from this table?

**Comment by ceorourke** [comment]
RIP lil table, we're going in a different direction.

**Pull request closed without merging**